spring-cloud-stream-integration-tests/src/test/java/org/springframework/cloud/stream/config/aggregate/AggregateApplicationTests.java

```diff
@@ -41,25 +42,28 @@
  * @author Oleg Zhurakousky
  */
 @RunWith(SpringJUnit4ClassRunner.class)
+@DirtiesContext
 public class AggregateApplicationTests {
 
-	@Before
-	public void before() {
-		System.setProperty("server.port", "0");
```

Comment:
Don't we still need this?

Reply:
@garyrussell I think this property was probably a superfluous one. No tests in the class failed after removing it. All the individual test contexts are getting cleaned up properly and don't think the individual tests will conflict. It could have been a carryover from previous times while it was failing for other reasons.

Reply:
Well, the tests will pass if it starts a web container and nothing else is listening on port 8080.

The reason we set this to 0 is so the operating system picks an unused port.

It can safely be removed as long as you are sure the test doesn't run a web container.

Reply:
Given that [web and actuator are actually optional](https://docs.spring.io/spring-cloud-stream/docs/Fishtown.BUILD-SNAPSHOT/reference/htmlsingle/#spring-cloud-stream-preface-actuator-web-dependencies) (one has to opt in), I say remove it.

Reply:
I believe:
```
<dependency>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-starter-actuator</artifactId>
	<optional>true</optional>
</dependency>
```
In the `spring-cloud-stream` POM does the trick for us and we really have a WEB in our test case. So, I would say remove, but change it into the explicit web environment exclusion (`web(false)` ?).

If we add something into the system properties, we  need to clean up it afterwards, otherwise the next test in the chain is going to reuse them. Just because it still the same JVM...
And that's why would be better to have a `@DirtiesContext` on all the `@SpringBootTest` to clean up all the application context related to this class.

Reply:
But, yeah... That `@DirtiesContext` could be as a separate story, indeed.

Reply:
BTW. This was always redundant in this test-case - we use there for application context a `web(false)` option. 🤷‍♂️